### PR TITLE
[MDEP-965] javax.json --> jakarta.json

### PR DESCRIPTION
**Pull request by elharo**
javax.json  is no longer supported. It has been replaced by jakarta.json. This shift is a little messy but only affects one test method in one test class.

### CODE REVIEW

**Review by michael-o — COMMENTED** [review]
I don't see the benefit here, this is just testing the output, no?

**Comment by elharo** [comment]
The issue is that javax.json hasn't been updated for about 5 years and won't be. It is an abandoned and unsupported library. It has been replaced by jakarta.json which has had multiple releases over those years. Something to do with the Eclipse Foundation taking over JEE development. But all usages of javax.json should be replaced with jakarta.json, which is the new hotness. Same library, same API but new package name and group ID.

File: src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo.java [hunk 1]
```diff
@@ -190,6 +189,7 @@ public void testTreeJsonCircularDependency() throws IOException {
     public void testTreeJsonParsing() throws Exception {
         List<String> contents = runTreeMojo("tree2.json", "json");
 
+        System.setProperty("jakarta.json.provider", "org.glassfish.json.JsonProviderImpl");
```
Comment (michael-o) [inline src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo.java:192]: Why is this necessary? Is there not `ServiceProvider` file in `META-INF?

Comment (elharo) [reply src/test/java/org/apache/maven/plugins/dependency/tree/TestTreeMojo.java:192]: Right now there doesn't seem to be. I can add it if you prefer but that will clutter the main jar with details only needed for the test.

File: pom.xml [hunk 2]
```diff
@@ -311,15 +311,15 @@ under the License.
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>javax.json</groupId>
-      <artifactId>javax.json-api</artifactId>
-      <version>1.1.4</version>
+      <groupId>jakarta.json</groupId>
+      <artifactId>jakarta.json-api</artifactId>
+      <version>2.1.3</version>
```
[the remark this answers is not in the document]
Comment (elharo) [reply pom.xml:316]: good catch. Looks like 2.0.2 should work.

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1471
